src/ray/core_worker/task_execution.h

```diff
@@ -0,0 +1,31 @@
+#ifndef RAY_CORE_WORKER_TASK_EXECUTION_H
+#define RAY_CORE_WORKER_TASK_EXECUTION_H
+
+#include "common.h"
+#include "ray/common/io.h"
+
+namespace ray {
+
+class CoreWorkerTaskExecutionInterface {
+ public:
+  CoreWorkerTaskExecutionInterface(CoreWorker *core_worker_)
+      : core_worker_(core_worker) {}
+
+  /// The callback provided app-language workers that executes tasks.
+  ///
+  /// \param ray_function[in] Information about the function to execute.
+  /// \param args[in] Arguments of the task.
+  /// \return Status.
+  using TaskExecutor =
+      std::function<Status(const RayFunction &ray_function, const vector<Buffer> &args)>;
+
+  /// Start receving and executes tasks in a infinite loop.
+  void StartWorker(const TaskExecutor &executor);
+
+ private:
+  CoreWorker *core_worker_;
```

Comment:
This should be a shared_ptr (as above, there won't be overhead associated with it, because it is only created/deleted very rarely)